courses/gnatmetric/010_gnatmetric.rst

```diff
@@ -0,0 +1,352 @@
+************
+GNATmetric
+************
+
+..
+    Coding language
+
+.. role:: ada(code)
+    :language: Ada
+
+.. role:: C(code)
+    :language: C
+
+.. role:: cpp(code)
+    :language: C++
+
+..
+    Math symbols
+
+.. |rightarrow| replace:: :math:`\rightarrow`
+.. |forall| replace:: :math:`\forall`
+.. |exists| replace:: :math:`\exists`
+.. |equivalent| replace:: :math:`\iff`
+
+..
+    Miscellaneous symbols
+
+.. |checkmark| replace:: :math:`\checkmark`
+
+==============
+Introduction
+==============
+
+------------------------
+Overview of gnatmetric
+------------------------
++ GNAT software metrics generation tool
++ Configurable metrics selection
++ Configurable scope of analysis
+
+  + All source files in the project
+  + One source file in the project
+  + A specific entity in a source file, e.g. a subprogram
+
+    + When applicable
+
++ :toolname:`GNAT Studio` provides a GUI interface
+
+  + Selecting the metrics to compute
+  + Selecting the scope of analysis
+  + Displaying the results
+
+-----------------------
+Invoking the Analyses
+-----------------------
+From the :menu:`Analyze` menu
+
+.. image:: ../../images/gnatmetric/menu_cascade.jpg
+
+---------------------
+Analysis Dialog Box
+---------------------
+
+.. image:: ../../images/gnatmetric/execute_dialog.jpg
+
+For this module, we will select *All line metrics*, *All complexity metrics*, *All syntax element metrics*, and *All coupling metrics*
+
+=======================
+Exploring the Results
+=======================
+
+-----------------
+Project Results
+-----------------
+
+Select project name (or GPR file) to get results across project
+
+.. image:: ../../images/gnatmetric/project_results.jpg
+
+----------------------
+Package Spec Results
+----------------------
+
+Select specification file to get results for that particular file
+
+.. image:: ../../images/gnatmetric/spec_results.jpg
+
+----------------------
+Package Body Results
+----------------------
+
+Select body file to get results for that particular file
+
+.. image:: ../../images/gnatmetric/body_results.jpg
+
+------------------------
+Line Metrics Explained
+------------------------
+
+**Average Lines In Body**
```

Comment:
Nit: those are definitions, for which we do have a `:dfn:` decorator now, if this decorator is a bit too "bright" to be used repeatedly we could reduce its brightness and replace it by a brighter `:term:` when needed (terms are included in a glossary in Sphinx, something cool to think about for our own use :) )